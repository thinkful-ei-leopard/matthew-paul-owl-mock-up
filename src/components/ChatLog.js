import React from "react";
import ChatMessage from "./ChatMessage";

function ChatLog(props) {
  return (
    <div className="Chat-Log">
        <div className="Message-Container">
            {props.chatEvents.map(event => 
                <ChatMessage 
                key={event.participantId}
                participantId={event.participantId}
                type={event.type}
                message={event.message}
                timestamp={event.timestamp}
                />
            )}
        </div>
        <div className="Chat-Box">
            <form id="Message-Form">
                <input type="text" placeholder="Chat"></input>
            </form>
        </div>
    </div>
  );
}

export default ChatLog;