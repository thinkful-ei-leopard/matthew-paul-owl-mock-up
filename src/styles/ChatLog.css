.Chat-Log {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  width: 300px;
  height: 100vh;
  padding: 20px;
  position: absolute;
  text-align: center;
  overflow-y: hidden;
  border: 2px solid black;
  vertical-align: baseline;
}

.Message-Container {
  position: relative;
  bottom: 0;
  margin-bottom: 20px;
}

.Chat-Box {
  display: block;
  width: 100%
}

.Chat-Box input {
  width: 100%;
  height: 40px;
}

.Chat-Action {
  text-align: center;
  padding: 5px;
  color: #888;
}

.Chat-Entry {
  text-align: left;
  padding: 5px;
}

.Chat-Message {
  width: 240px;
  text-align: center;
}

.Chat-Timestamp {
  margin-left: 5px;
  color: #888;
}