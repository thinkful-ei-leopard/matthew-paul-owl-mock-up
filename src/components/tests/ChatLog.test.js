import React from 'react';
import ReactDOM from 'react-dom';
import ChatLog from '../ChatLog';
import STORE from '../../STORE';
import renderer from 'react-test-renderer'; 

it('renders without crashing', () => {
  // first create a DOM element to render the component into
  const div = document.createElement('div');

  // render the component, this is the actual test, if something is wrong it will fail here
  ReactDOM.render(<ChatLog chatEvents={STORE.chatEvents}/>, div);

  // clean up code
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
  .create(<ChatLog chatEvents={STORE.chatEvents}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();  
});