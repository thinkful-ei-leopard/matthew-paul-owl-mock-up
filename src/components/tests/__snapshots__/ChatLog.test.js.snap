// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="Chat-Log"
>
  <div
    className="Message-Container"
  >
    <div
      className="Chat-Entry"
    >
      <img
        alt="Avatar"
        className="Participant-Avatar"
        src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
      />
      <span
        className="Participant-Name"
      >
        Axe Kubicka
      </span>
      <span
        className="Chat-Timestamp"
      >
        1548852484247
      </span>
      <p
        className="Chat-Message"
      >
        Hello world
      </p>
    </div>
    <div
      className="Chat-Entry"
    >
      <span
        className="Chat-Action"
      >
        Caty Flucker
         gave a thumbs up!
      </span>
    </div>
    <div
      className="Chat-Entry"
    >
      <span
        className="Chat-Action"
      >
        Frank Runciman
         gave a thumbs down!
      </span>
    </div>
    <div
      className="Chat-Entry"
    >
      <span
        className="Chat-Action"
      >
        Koren Templeton
         raised their hand!
      </span>
    </div>
    <div
      className="Chat-Entry"
    >
      <span
        className="Chat-Action"
      >
        Koren Templeton
         clapped!
      </span>
    </div>
    <div
      className="Chat-Entry"
    >
      <span
        className="Chat-Action"
      >
        Koren Templeton
         joined!
      </span>
    </div>
    <div
      className="Chat-Entry"
    >
      <span
        className="Chat-Action"
      >
        Ashla Attwool
         left!
      </span>
    </div>
    <div
      className="Chat-Entry"
    >
      <span
        className="Chat-Action"
      >
        Axe Kubicka
         joined the stage!
      </span>
    </div>
    <div
      className="Chat-Entry"
    >
      <span
        className="Chat-Action"
      >
        Axe Kubicka
         left the stage!
      </span>
    </div>
  </div>
  <div
    className="Chat-Box"
  >
    <form
      id="Message-Form"
    >
      <input
        placeholder="Chat"
        type="text"
      />
    </form>
  </div>
</div>
`;
